File: 15-key-vault/python-sdk/proxy_keys.py
"""
Cost Katana Key Vault: Proxy Keys (Python)

Run: python 19-key-vault/python-sdk/proxy_keys.py
"""

import os
import requests

API_BASE = "https://cost-katana-backend.store/api"
API_KEY = os.getenv("COST_KATANA_API_KEY")

def create_proxy_key(config):
    response = requests.post(
        f"{API_BASE}/key-vault/proxy-keys",
        json=config,
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print(f"✅ Created proxy key: {response.json()['data']['proxyKey']}")
    return response.json()["data"]

def list_proxy_keys():
    response = requests.get(
        f"{API_BASE}/key-vault/proxy-keys",
        headers={"Authorization": f"Bearer {API_KEY}"}
    )
    response.raise_for_status()
    print(f"📋 Found {len(response.json()['data'])} proxy keys")
    return response.json()["data"]

if __name__ == "__main__":
    print("🥷 Key Vault: Proxy Keys\n")
    create_proxy_key({
        "name": "Frontend Key",
        "permissions": ["chat"],
        "rateLimit": {"requestsPerMinute": 60},
        "budgetLimit": 100
    })
    list_proxy_keys()

File: 15-key-vault/npm-package/proxy-keys.ts
/**
 * Cost Katana Key Vault: Proxy Keys
 * 
 * Run: npx ts-node 19-key-vault/npm-package/proxy-keys.ts
 */

import axios from 'axios';

const API_BASE = 'https://api.costkatana.com/api'\;
const API_KEY = process.env.COST_KATANA_API_KEY;

async function createProxyKey(config: any) {
  const response = await axios.post(
    `${API_BASE}/key-vault/proxy-keys`,
    config,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log(`✅ Created proxy key: ${response.data.data.proxyKey}`);
  return response.data.data;
}

async function listProxyKeys() {
  const response = await axios.get(
    `${API_BASE}/key-vault/proxy-keys`,
    { headers: { 'Authorization': `Bearer ${API_KEY}` } }
  );
  console.log(`📋 Found ${response.data.data.length} proxy keys`);
  return response.data.data;
}

async function main() {
  console.log('🥷 Key Vault: Proxy Keys\n');
  await createProxyKey({
    name: 'Frontend Key',
    permissions: ['chat'],
    rateLimit: { requestsPerMinute: 60 },
    budgetLimit: 100
  });
  await listProxyKeys();
}

if (require.main === module) main();
export { createProxyKey, listProxyKeys };
